torch/csrc/jit/python/script_init.cpp

```diff
@@ -846,6 +846,12 @@ void initJitScriptBindings(PyObject* module) {
     }
   }
 
+  object_class.def(
+      "__deepcopy__", [](const Object& self, const py::dict& memo) {
+        // TODO: Figure out if memoization makes any sense in this context
```

Comment:
It does: if two script objects exist as a separate part of the model but share state, this will break that sharing. It's probably not frequent, but it is one of those things that would be bad to leave waiting for someone to hit. If you stash the memo table for IValue's deep copy in the memo dictionary under some known key, then it can be used across deep copy calls in IValue.